Misc/NEWS.d/next/C_API/2025-09-10-17-49-54.gh-issue-138762.1rG6GQ.rst

```diff
@@ -0,0 +1 @@
+Move Include/dynamic_annotations.h to be outside extern "C"
```

Comment:
Please, describe the effect of this change, not the change itself. My suggestion:


```suggestion
Fix linking of the free-threaded build with C++ compiler.
```